chore(client): Add status code types

rozzilla opened the pull request:
Extremely useful to better narrow down the types to the proper status code (f.e. in [this case](https://github.com/platformatic/platformatic/issues/2213))

mcollina reviewed (commented):
How are those supposed to be used? Can you add some type tests?

I think https://github.com/fastify/fastify/blob/0685fc7b1fc95055c459f3a557e2b99332288e1c/types/utils.d.ts#L56 might be a better approach to generate the status codes.

rozzilla commented:
> How are those supposed to be used?
I added `exampleUsageOfStatusCodeType` in the tests, to make it more clear. As you can see from this example fn after a check `status` type equals `StatusCode2xx`. The goal is to narrow down the type to a proper status code, so that for instance when `@platformatic/client` returns a `FullResponse` for a `200` status code, I can more easily access this type.

> Can you add some type tests?
Done on d87224b ✅ 

> I think https://github.com/fastify/fastify/blob/0685fc7b1fc95055c459f3a557e2b99332288e1c/types/utils.d.ts#L56 might be a better approach to generate the status codes
I love that approach: simple, elegant and effective! Updated leveraging the `fastify` types 👍🏼

mcollina reviewed (approved):
lgtm

mcollina commented:
cc @leorossi
